opm/input/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -814,6 +777,52 @@ bool FieldProps::has<int>(const std::string& keyword) const {
 }
 
 
+void FieldProps::apply_multipliers()
+{
+    bool hasPorvBefore = (this->double_data.find(ParserKeywords::PORV::keywordName) == this->double_data.end());
+    static const auto prefix = getMultiplierPrefix();
+
+    for(const auto& [mult_keyword, kw_info]: multiplier_kw_infos_)
+    {
+        const std::string keyword = mult_keyword.substr(prefix.size());
+        auto mult_iter = this->double_data.find(mult_keyword);
+        assert(mult_iter != this->double_data.end());
+        auto iter = this->double_data.find(keyword);
+        if (iter == this->double_data.end()) {
+            iter = this->double_data
+                .emplace(std::piecewise_construct,
+                         std::forward_as_tuple(keyword),
+                         std::forward_as_tuple(kw_info, this->active_size, kw_info.global ? this->global_size : 0))
+                .first;
+        }
+        using Scalar = typename std::remove_cv_t<std::remove_reference_t<decltype(iter->second.data[0])>>;
+        std::transform(iter->second.data.begin(), iter->second.data.end(),
+                       mult_iter->second.data.begin(), iter->second.data.begin(),
+                       std::multiplies<Scalar>());
```

Comment:
I'll mention once more that this code would be simpler if you did not explicitly specify the `Scalar` type and instead passed `std::multiplies<>()` (literally) as the final argument to `std::transform()`.  Using the defaulted template parameter (formally `void`) means we get template argument deduction on the parameters passed to the function call operator of `std::multiplies<>` and that is usually what we want.